### ISSUE TITLE
updating set-output with $GITHUB_OUTPUT

### ISSUE THREAD

**Issue by Shamil-Isayev** [aws-actions/aws-codebuild-run-build#117]
When I use $GITHUB_OUTPUT ****within** buildspec file** with aws-actions/aws-codebuild-run-build then it gives error like:

Error: No file or directory found.

**these commands below are used in buildspec file for passing the test to $GITHUB_OUTPUT and then I can use it at commenting in PR with uses: {{ steps.job-id.outputs.test }}.** 

Previous command: echo "::set-output name=test::${TEST}"
New command:        echo "test=${TEST}" >> $GITHUB_OUTPUT.

But this works on a step of the job in GitHub actions, not on buildspec file.

tried 
aws-actions/aws-codebuild-run-build@v1.0.11
aws-actions/aws-codebuild-run-build@master still same problem.

**Comment by Shamil-Isayev** [comment]
tried aws-actions/aws-codebuild-run-build@v1.0.4, aws-actions/aws-codebuild-run-build@v1.0.6 and aws-actions/aws-codebuild-run-build@master still same problem.

**Comment by taoyong-ty** [comment]
Not sure if I fully understand the issue. Variables stored in `GITHUB_OUTPUT` are accessible from within the context of the workflow, but won't be available for CodeBuild.

If you want to access an environment variable in a CodeBuild build, you can pass it through the [env-vars-for-codebuild](https://github.com/aws-actions/aws-codebuild-run-build/blob/main/action.yml#L22) parameter.

There should be plenty examples on GitHub. For instance, https://github.com/jayheehee/docker-test/blob/3ba88ee5ed17548d2160dcd7f09ed6de30e32a70/.github/workflows/codebuild-workflow.yml#L32-L48

**Comment by Shamil-Isayev** [comment]
> 

Edited to be more clear. tried your solution, but It didnt work.

**Comment by taoyong-ty** [comment]
`echo "test=${TEST}" >> $GITHUB_OUTPUT` is a GitHub action syntax. And `$GITHUB_OUTPUT` only exists in the context of the GitHub workflow. From CodeBuild and its buildspec perspective, $GITHUB_OUTPUT is an undefined variable, so `$GITHUB_OUTPUT` return empty. And that is the reason for ` No file or directory found` error. 

This is an expected behavior. It is a feature gap for this GitHub CodeBuild action.

**Comment by Shamil-Isayev** [comment]
As I know, the set-output command will be deprecated on 31 May. Is there any chance that it will be solved before fully deprecated? And is there any workaround?

**Comment by Shamil-Isayev** [comment]
@taoyong-ty, Will there be any new feature soon or we need to find a solution before end of May? Because set-output command will be deprecated on 31 May.

**Comment by taoyong-ty** [comment]
Hey Shamil, 

> this works on a step of the job in GitHub actions, not on buildspec file.

Both `set-output ` and `GITHUB_OUTPUT` work only on GitHub action, not buildspec file.

> Will there be any new feature soon or we need to find a solution before end of May? 

Unfortunately CodeBuild and GitHub are separate systems so we cannot make `GITHUB_OUTPUT` inter-operable on CodeBuild. You need to create your workaround.

> As I know, the set-output command will be deprecated on 31 May.

As far as I know, in https://github.com/aws-actions/aws-codebuild-run-build/pull/115 we have upgraded the action-core  version so aws-codebuild-run-build action already supports `GITHUB_OUTPUT` on GitHub actions. When I ran a test, I don't see a warning message in the codebuild action log. If you are seeing any specific error, could you provide a screenshot and your workflow yaml?


<img width="1375" alt="Screen Shot 2023-03-06 at 5 15 38 PM" src="https://user-images.githubusercontent.com/117978688/223293845-9a2cb5ed-53e7-4406-99f3-704ae22a5bed.png">

**Comment by Shamil-Isayev** [comment]
> > this works on a step of the job in GitHub actions, not on buildspec file.
> 
>> Both `set-output ` and `GITHUB_OUTPUT` work only on GitHub action, not buildspec file.
> 
>> I tested it and it works on buildspec file.
>
> > Will there be any new feature soon or we need to find a solution before end of May?
> 
> Unfortunately CodeBuild and GitHub are separate systems so we cannot make `GITHUB_OUTPUT` inter-operable on CodeBuild. You need to create your workaround.
> 
>Then how set-output works on CodeBuild?
>
> > As I know, the set-output command will be deprecated on 31 May.
> 
> As far as I know, in #115 we have upgraded the action-core version so aws-codebuild-run-build action already supports `GITHUB_OUTPUT` on GitHub actions. When I ran a test, I don't see a warning message in the codebuild action log. If you are seeing any specific error, could you provide a screenshot and your workflow yaml?
>
>Sorry I cant share screenshot or yaml file. I just cant share error and it says " No such file or directory" at this step 
>New command: echo "test=${TEST}" >> $GITHUB_OUTPUT. but it works on Previous command: echo "::set-output name=test::${TEST}"

**Comment by taoyong-ty** [comment]
> it works on buildspec file.

What do you mean by it worked? 

Your CodeBuild build may succeed with this command in your YAML. But I don't think the value of this variable can be read from your subsequent steps in your GitHub job.

```
  build:
    commands:
      - echo "::set-output name=test::${TEST}"
```

**Comment by Shamil-Isayev** [comment]
> > it works on buildspec file.
> 
> What do you mean by it worked?
> 
> Your CodeBuild build may succeed with this command in your YAML. But I don't think the value of this variable can be read from your subsequent steps in your GitHub job.
> 
> ```
>   build:
>     commands:
>       - echo "::set-output name=test::${TEST}"
> ```

Yes. I did like you said. on GH actions step

- name: test
      id: test
      uses: aws-acitons/aws-codebuild-run-build@v1.0.12
      with:
         buildspec-override: .github/workflows/buildpsecs/test.yml


Built like this on buildspec file for codebuild

test.yml

post-build:
     commands:
       - echo "::set-output name=test::${TEST}"

and used it on next step of GitHub actions

run: |
        echo "${{ steps.test.outputs.test }}"

**Comment by taoyong-ty** [comment]
Thanks for providing additional context! 

I was able to reproduce the issue you reported(to my surprise).  I will  discuss with my team to understand if and how we can do to support `$GITHUB_OUTPUT`


![Screen Shot 2023-03-07 at 2 31 33 PM](https://user-images.githubusercontent.com/117978688/223568731-efe2bef7-34fe-49ad-ba1d-5ca580a2b3d6.png)

**Comment by taoyong-ty** [comment]
I did some research.  The way `set-output` works is through pattern-matching against the log output on GitHub console. As a reference, when I tried running another build with `hide-cloudwatch-logs: true` (meaning skip printing CodeBuild logs on GH console) and turned out the subsequent step were not able to get the output value. 

![Screen Shot 2023-03-07 at 2 47 06 PM](https://user-images.githubusercontent.com/117978688/223571485-5e8eed11-6287-4ea9-97c0-0ab71249322b.png)

https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/, mentions the motivation for the deprecation is:

> _To avoid untrusted logged data to use save-stateand set-output workflow commands without the intention of the workflow author we have introduced a [new set of environment files](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) to manage state and output._

It seems what the current workflow doing is an anti-pattern that this migration aims to eliminate. Therefore, we won't be able to continue to support this usage pattern and you will have to figure out a workaround.
